scripts/js/lib/api/releaseNotes.ts

```diff
@@ -257,3 +272,53 @@ async function writeReleaseNoteForVersion(
     await writeFile(versionPath, `${initialHeader}\n## ${versionsMarkdown}`);
   }
 }
+
+export function groupByMajorVersion(
+  versions: string[],
+): Record<string, string[]> {
+  const grouped = groupBy(versions, (v) => {
+    const match = v.match(/\[(\d+\.\d+)\]/);
+    if (!match || !match[1]) return "unknown";
```

Comment:
I'd rather that we throw an exception. We want to know that the input doesn't match our expectations. Ditto line 284.